src/checkers/inference/solver/backend/z3smt/Z3SmtSolver.java

```diff
@@ -0,0 +1,429 @@
+    private void parseStdOut(BufferedReader stdOut, List<String> results) {
+        String line = "";
+
+        boolean declarationLine = true;
+        // each result line is "varName value"
+        String resultsLine = "";
+
+        boolean unsat = false;
+
+        try {
+            while ((line = stdOut.readLine()) != null) {
+                line = line.trim();
+
+                if (getUnsatCore) {
+                    // UNSAT Cases ====================
+                    if (line.contentEquals("unsat")) {
+                        unsat = true;
+                        continue;
+                    }
+                    if (unsat) {
+                        if (line.startsWith("(")) {
+                            line = line.substring(1); // remove open bracket
+                        }
+                        if (line.endsWith(")")) {
+                            line = line.substring(0, line.length() - 1);
+                        }
+
+                        for (String constraintID : line.split(" ")) {
+                            unsatConstraintIDs.add(constraintID);
+                        }
+                    }
+                } else {
+                    // SAT Cases =======================
+                    // processing define-fun lines
+                    if (declarationLine && line.startsWith("(define-fun")) {
+                        declarationLine = false;
+
+                        int firstBar = line.indexOf('|');
+                        int lastBar = line.lastIndexOf('|');
+
+                        assert firstBar != -1;
+                        assert lastBar != -1;
+                        assert firstBar < lastBar;
+                        assert line.contains("Bool") || line.contains("Int");
+
+                        // copy z3 variable name into results line
+                        resultsLine += line.substring(firstBar + 1, lastBar);
+                        continue;
+                    }
+                    // processing lines immediately following define-fun lines
+                    if (!declarationLine) {
+                        declarationLine = true;
+                        String value = line.substring(0, line.lastIndexOf(')'));
+
+                        if (value.contains("-")) { // negative number
+                            // remove brackets surrounding negative numbers
+                            value = value.substring(1, value.length() - 1);
+                            // remove space between - and the number itself
+                            value = String.join("", value.split(" "));
+                        }
+
+                        resultsLine += " " + value;
+                        results.add(resultsLine);
+                        resultsLine = "";
+                    }
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    void printArithmeticConstraints() {
```

Comment:
Unused method? Remove? Make private and add TODO when to call?

Reply:
This might be useful for debugging? What's the typical way to deal with code only for debugging?